dynamic/discovery/discovery.go

```diff
@@ -131,6 +149,8 @@ func (rm *ResourceMap) refresh() {
 			// Make sure we don't choose a subresource like "pods/status".
 			if !strings.ContainsRune(apiResource.Name, '/') {
 				gve.kinds[apiResource.Kind] = apiResource
+			} else {
+				gve.subresources[apiResource.Name] = apiResource
```

Comment:
After you store all these subresources, do another pass over all subresources to denormalize the info into the corresponding main resource, so that the main APIResource contains a list of subresources and a `HasSubresource` method to query it.

Reply:
@enisoc 
Another alternative is to denormalize the info on the spot without introducing the another pass. As `gve` maps are of `map[string]*APIResource` type, the change will be reflected immediately (the `APIResource` is a pointer), I think the another pass should be unnecessary.

Do you have any reason for introducing the another pass (e.g., better code readability)?

Reply:
What we want is to add some data to the APIResource for "foo" while we are processing the entry for "foo/status". The second pass is necessary to make sure that "foo" has already been processed when we look for it to add the subresource data. Even if it's technically the case that the discovery API always returns "foo" before "foo/status" in the list (which I have no reason to suspect anyway), it would be brittle to depend on that ordering.

Reply:
I think we should store all subresources, not just `/status`. This fits better with `HasSubresource("someSubresource")` function.